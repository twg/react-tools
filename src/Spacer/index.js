import React from 'react'
import css from './style.css'
import PropTypes from 'prop-types'

export const Spacer = props => {
  const classes = [css.spacer, props.size ? css[props.size] : ''].join(' ')

  return <div className={classes} />
}

Spacer.propTypes = {
  /**
   * Available values: small, large
   */
  size: PropTypes.string
}

export default Spacer
