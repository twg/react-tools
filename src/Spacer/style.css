.spacer {
    width: 100%;
    height: 40px;
}

.small{
    height: 20px;
}

.large{
    height: 60px;
}
