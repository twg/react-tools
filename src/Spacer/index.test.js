/* eslint-env jest */

import React from 'react'
import Spacer from './'
import renderer from 'react-test-renderer'
import cases from 'jest-in-case'

cases(
  'matches snapshots',
  opts => {
    expect(
      renderer.create(<Spacer {...opts.props} />).toJSON()
    ).toMatchSnapshot()
  },
  [
    { name: 'normal', props: { size: 'normal' } },
    { name: 'small', props: { size: 'small' } },
    { name: 'large', props: { size: 'large' } }
  ]
)
