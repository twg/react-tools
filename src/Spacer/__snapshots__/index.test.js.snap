// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots large 1`] = `
<div
  className="spacer large"
/>
`;

exports[`matches snapshots normal 1`] = `
<div
  className="spacer normal"
/>
`;

exports[`matches snapshots small 1`] = `
<div
  className="spacer small"
/>
`;
